deploy/template.yaml

```diff
@@ -1125,6 +1137,175 @@ Resources:
       DatapointsToAlarm: 1
       Threshold: 1
       ComparisonOperator: GreaterThanOrEqualToThreshold
+  
+  HmrcTokenFunction:
+    Type: AWS::Serverless::Function
+    DependsOn:
+      - "HmrcTokenLogGroup"
+    Properties:
+      FunctionName: !Sub "HmrcToken-${AWS::StackName}"
+      CodeUri: ../src/
+      Handler: HmrcTokenHandler.lambdaHandler
+      ReservedConcurrentExecutions: !If
+        - ApplyReservedConcurrency
+        - !Ref LambdaConcurrency
+        - !Ref AWS::NoValue
+      # Events:        
+      #   InvocationLevel:
+      #     Type: Schedule
+      #     Properties:
+      #       Schedule: cron(0/5 * * * ? *)          
+      Environment:
+        Variables:
+          HMRC_CLIENT_ID_SSM_PATH: !Sub "/${Environment}/HMRC/CLIENT_ID"
+          HMRC_CLIENT_SECRET_SSM_PATH: !Sub "/${Environment}/HMRC/CLIENT_SECRET"
+          HMRC_BASE_URL: !FindInMap [ EnvironmentVariables, !Ref Environment, HMRCBASEURL ]
+      Policies:
+        - AWSLambdaBasicExecutionRole
+        - AWSXrayWriteOnlyAccess
+        - SSMParameterReadPolicy:
+            ParameterName: !Sub "${Environment}/HMRC/CLIENT_ID"
+        - SSMParameterReadPolicy:
+            ParameterName: !Sub "${Environment}/HMRC/CLIENT_SECRET"
+        - Statement:
+            - Sid: SSMParameterWritePolicy
+              Effect: Allow
+              Action:
+                - ssm:PutParameter
+              Resource:
+                - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${Environment}/HMRC/TOKEN"
+    Metadata:
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        EntryPoints:
+          - HmrcTokenHandler.ts
+  
+  HmrcTokenScheduledRule: 
+    Type: AWS::Events::Rule
+    Condition: NotLocalTestStack
+    Properties: 
+      Description: "ScheduledRule"
+      ScheduleExpression: cron(0/5 * * * ? *)
+      State: "ENABLED"
+      Targets: 
+        - 
+          Arn: 
+            Fn::GetAtt: 
+              - "HmrcTokenFunction"
+              - "Arn"
+          Id: "HmrcTokenFunction"
+  # PermissionForEventsToInvokeLambda: 
```

Comment:
Is this meant to be commented out? Looks like it's needed to actually enable the cron?

Reply:
we have HmrcTokenFunctionPermission defined which gives permissions to all resources and this wont be required. I just commented it to test it in main stack before i confirm and remove it.

Reply:
Removed the commented lines.